README.md

```diff
@@ -35,686 +43,422 @@ mix.exs
-the config.
-
-```elixir
-# Fetch permissions from the claims map
+### Plugs out of the box
 
-Guardian.Permissions.from_claims(claims, :default)
-Guardian.Permissions.from_claims(claims, :admin)
+#### `Guardian.Plug.VerifyHeader`
+Look for a token in the header and verify it
 
-# Check the permissions for all present
+#### `Guardian.Plug.VerifySession`
+Look for a token in the session and verify it
 
-Guardian.Permissions.from_claims(claims, :default) |> Guardian.Permissions.all?([:read, :write], :default)
-Guardian.Permissions.from_claims(claims, :admin) |> Guardian.Permissions.all?([:reconcile], :admin)
+#### `Guardian.Plug.VerifyCookie`
+Look for a token in cookies and exchange it for an access token
 
-# Check for any permissions
-Guardian.Permissions.from_claims(claims, :default) |> Guardian.Permissions.any?([:read, :write], :default)
-Guardian.Permissions.from_claims(claims, :admin) |> Guardian.Permissions.any?([:reconcile, :dashboard], :admin)
-```
+#### `Guardian.Plug.EnsureAuthenticated`
+Make sure that a token was found and is valid
 
-You can use a plug to ensure permissions are present. See Guardian.Plug.EnsurePermissions
+#### `Guardian.Plug.EnsureNotAuthenticated`
+Make sure no one is logged in
 
-#### Setting permissions
+#### `Guardian.Plug.LoadResource`
+If a token was found, load the resource for it
 
-When you generate (or sign in) a token, you can inject permissions into it.
+See the documentation for each Plug for more information.
 
-```elixir
-Guardian.encode_and_sign(resource, :access, perms: %{ admin: [:dashboard], default: Guardian.Permissions.max}})
-```
-
-By setting a permission using Guardian.Permission.max you're setting all the bits, so even if new permissions are added, they will be set.
+### Pipelines
 
-You can similarly pass a `:perms` key to the sign\_in method to have the
-permissions encoded into the token.
+A pipeline is a way to collect together the various plugs for a particular authentication scheme.
 
-### Hooks
+Apart from keeping an authentication flow together, pipelines provide downstream information for error handling and which token module to use. You can provide this separately but we recommend creating a pipeline plug.
 
-Often you'll need to take action on some event within the lifecycle of
-authentication. Recording logins etc. Guardian provides hooks to allow you to do
-this. Use the Guardian.Hooks module to setup. Default implementations are
-available for all callbacks.
+#### Create a custom pipeline
 
 ```elixir
-defmodule MyApp.GuardianHooks do
-  use Guardian.Hooks
+defmodule MyApp.AuthAccessPipeline do
+  use Guardian.Plug.Pipeline, otp_app: :my_app,
+                              module: MyApp.AuthTokens,
+                              error_handler: MyApp.AuthErrorHandler
 
-  def after_sign_in(conn, location) do
-    user = Guardian.Plug.current_resource(conn, location)
-    IO.puts("SIGNED INTO LOCATION WITH: #{user.email}")
-    conn
-  end
+  alias Guardian.Plug.{
```

Comment:
Let's remove these aliases and just put the full paths...easier to copy/paste for users.